killerbee/scapy_extensions.py

```diff
@@ -547,7 +551,9 @@ def kbencrypt(pkt, data, key = None, verbose = None):
         print "\tMic:            " + mic.encode('hex')
 
     # Set pkt's values to reflect the encrypted ones to it's ready to be sent
-    f['data'] = payload
-    f['mic'] = struct.unpack(">I", mic)[0]
+    # note that calculated mic is for the encrypted sublayer, not the enclosing packet
+    # TODO: calculate mic for overall packet if required
```

Comment:
This should become clearer in my next stage updates but I think it's worth merging now as NWK encryption is currently broken without these patches.

For the next stage I will be updating APS layer crypto which needs to take into account key type as well as nwk_seclevel (at NWK level it is always fixed at AES CCM with 32-bit MIC, but at layers after NWK it could be anything).

Reply:
Ok - that makes sense. Let me know when this is good to merge and I'll be happy to -- thanks for working on this!

Reply:
No worries - you can pull the trigger now if you're happy with current state.